src/zon.zig

```diff
@@ -481,47 +489,66 @@ const Parser = struct { // MARK: Parser
 	/// Assumes that the region starts with a number character ('+', '-', '.' or a digit).
 	fn parseNumber(chars: []const u8, index: *u32) ZonElement {
 		var sign: i2 = 1;
+		const numberStartIndex = index.*;
+
 		if(chars[index.*] == '-') {
 			sign = -1;
 			index.* += 1;
 		} else if(chars[index.*] == '+') {
 			index.* += 1;
 		}
-		var intPart: i64 = 0;
-		if(index.* + 1 < chars.len and chars[index.*] == '0' and chars[index.* + 1] == 'x') {
-			// Parse hex int
+		var subString: []const u8 = undefined;
+		if(chars.len > 2 and chars[index.*] == '0' and (chars[index.* + 1] == 'x' or chars[index.* + 1] == 'b')) {
 			index.* += 2;
 			while(index.* < chars.len) : (index.* += 1) {
 				switch(chars[index.*]) {
-					'0', '1', '2', '3', '4', '5', '6', '7', '8', '9' => {
-						intPart = (chars[index.*] - '0') +% intPart*%16;
-					},
-					'a', 'b', 'c', 'd', 'e', 'f' => {
-						intPart = (chars[index.*] - 'a' + 10) +% intPart*%16;
-					},
-					'A', 'B', 'C', 'D', 'E', 'F' => {
-						intPart = (chars[index.*] - 'A' + 10) +% intPart*%16;
+					'0', '1', '2', '3', '4', '5', '6', '7', '8', '9' => {},
+					'a',
+					'b',
+					'c',
+					'd',
+					'e',
+					'f',
+					=> {},
```

Comment:
```suggestion
					'a',
					'b',
					'c',
					'd',
					'e',
					'f'
					=> {},
```

Remove trailing comma and reformat

Reply:
Doesn't zig like trailing commas?

Reply:
The key here is to reformat after removing the trailing commas.

Reply:
Oh i see